docs/sandwich-tests/index.md

```diff
@@ -0,0 +1,383 @@
+switches](https://opam.ocaml.org/blog/opam-local-switches/), which are similar
+to Node project directories[^2]. You can list all the opam switches on your
+computer by running
+
+```shell
+opam switch
+```
+
+The output will look something like this:
+
+```
+#  switch                     compiler                   description
+→  ~/melange-for-react-devs   ocaml-base-compiler.5.1.1  ~/melange-for-react-devs
+   default                    ocaml.5.1.0                default
+
+[NOTE] Current switch has been selected based on the current directory.
+       The current global system switch is default.
+```
+
+As implied by `[NOTE]`, you don't need to manually set the opam switch for your
+project, the switch is set based on your current working directory[^3].
+
+## First test
+
+Add a new file `src/order-confirmation/SandwichTests.re` and add a simple test
+to it:
+
+<<< SandwichTests.re#first-test
+
+You should get this error:
+
+```
+File "docs/burger-tests/SandwichTests.re", line 2, characters 0-9:
+2 | Fest.test("Item.Sandwich.toEmoji", () =>
+    ^^^^^^^^^
+Error: Unbound module Fest
+```
+
+In order to use a library, you must add it to the `libraries` field of the
+`melange.emit` stanza in your `dune` file:
+
+```clj{3}
+(melange.emit
+ (target output)
+ (libraries reason-react melange-fest)
+ (preprocess
+  (pps melange.ppx reason-react-ppx))
+ (module_systems es6)
+ (runtime_deps
+  (glob_files *.css)))
+```
+
+## Opening a module
+
+We'll be adding several more tests to this file, and it'll quickly become
+tiresome to have to write `Fest.test`, `Fest.expect`, and `Fest.equal` all the
+time.
+
+The easiest way to save typing is by using a module alias:
+
+<<< SandwichTests.re#module-alias
+
+However, because `SandwichTests` is a module for tests and `Fest` only contains
+testing-related functions, it's reasonable to
+[open](https://reasonml.github.io/docs/en/module#opening-modules) the `Fest`
+module and make all its functions available inside the scope of the
+`SandwichTests` module:
+
+<<< SandwichTests.re#open-fest
+
+::: warning
+
+In most cases, `open` should not be used at the toplevel of a module. It usually
+makes more sense to use a [local
+open](https://reasonml.github.io/docs/en/module#local-opens) which makes
+all the functions of the opened module available inside the scope of a function
+or submodule.
+
+:::
+
+## Compile with `.mjs` extension
+
+Now that your `SandwichTests.re` compiles, try running the outputted `.js` file
+in Node:
+
+```shell
+node _build/default/src/order-confirmation/output/src/order-confirmation/SandwichTests.js
+```
+
+You'll probably see this error:
+
+```
+(node:68498) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
+(Use `node --trace-warnings ...` to show where the warning was created)
+/home/fhsu/work/melange-for-react-devs/_build/default/src/order-confirmation/output/src/order-confirmation/SandwichTests.js:3
```

Comment:
Maybe this path could be simplified to `~/melange-for-react-devs-template/_build...` as we told users to download the template in the first steps of the book.

Reply:
I looked around, and I'm not at all consistent about what paths I use in the book. I'll address this in a separate PR.